### PR TITLE
Check for missing const-stability attributes in `rustc_passes`

### PR DESCRIPTION
**Pull request by ecstatic-morse**
Currently, this happens as a side effect of `is_min_const_fn`, which is non-obvious. Also adds a test for this case, since we didn't seem to have one before.

### CODE REVIEW

File: compiler/rustc_mir/src/const_eval/fn_queries.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ pub fn is_min_const_fn(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
             None => {
                 if let Some(stab) = tcx.lookup_stability(def_id) {
                     if stab.level.is_stable() {
-                        tcx.sess.span_err(
```
Comment (oli-obk) [inline compiler/rustc_mir/src/const_eval/fn_queries.rs:53]: can we just `span_bug!` here now, or is it still reachable even if the stability error was reported?

Comment (ecstatic-morse) [reply compiler/rustc_mir/src/const_eval/fn_queries.rs:53]: This function is called in `check_unsafety`, which runs before `MissingStabilityAnnotations`.

**Pull request merged**